Do not use paths from Fink, /usr/local, or other non-MacPorts locations.
--- configure.ac.orig	2021-02-17 12:02:32.000000000 -0600
+++ configure.ac	2021-03-08 04:51:56.000000000 -0600
@@ -112,9 +112,7 @@
 AC_SUBST([HOGWEED_EXTRA_SYMBOLS])
 
 LSH_RPATH_INIT([`echo $with_lib_path | sed 's/:/ /g'` \
-    `echo $exec_prefix | sed "s@^NONE@$prefix/lib@g" | sed "s@^NONE@$ac_default_prefix/lib@g"` \
-    /usr/local/lib /sw/local/lib /sw/lib \
-    /usr/gnu/lib /opt/gnu/lib /sw/gnu/lib /usr/freeware/lib /usr/pkg/lib])
+    `echo $exec_prefix | sed "s@^NONE@$prefix/lib@g" | sed "s@^NONE@$ac_default_prefix/lib@g"`])
 
 # Checks for programs.
 AC_PROG_CC
